pkg/verifier/safe_delete.go

```diff
@@ -31,23 +36,48 @@ func SafeDelete(ctx context.Context, logger log.Logger, bkt objstore.Bucket, bac
 		return errors.Errorf("%s dir seems to exists in backup bucket. Remove this block manually if you are sure it is safe to do", id)
 	}
 
-	tempdir, err := ioutil.TempDir("", "safe-delete")
-	if err != nil {
-		return err
-	}
-	dir := filepath.Join(tempdir, id.String())
-	err = os.Mkdir(dir, 0755)
-	if err != nil {
-		return err
+	if useExistingTempdir && tempdir == "" {
```

Comment:
It seems like they **both** can be set or unset which proves my point.

Reply:
I touched up the comments for this function again so that they better describe the behavior of the current code.  No code changes though.

Reply:
I proposed different approach - splitting this into 2 function which will solve this issue (:

Reply:
This is now committed.  The function signatures didn't quite work out as we would have liked, but I totally aggree, this feels much, much nicer.